Added section about preview-bundle

wachterjohannes opened the pull request:
| Q | A
| --- | ---
| Fixed tickets | none
| License | MIT

#### What's in this PR?

This PR add documentation about preview which was introduced by https://github.com/sulu/sulu/pull/2332

danrot commented on bundles/preview/index.rst:
```diff
@@ -0,0 +1,117 @@
+=============
+
+The PreviewBundle provides the feature of preview for custom-entities.
+It is build ontop the route-bundle and can only be introduced for
```
`on top of the route-bundle`

danrot commented on bundles/preview/index.rst:
```diff
@@ -0,0 +1,117 @@
+entities which has an `RouteDefaultsProvider`.
+
+This Provider will be used to determine which controller should be
+used to render the html of the entity. This html should contain the
```
`HTML`

danrot commented on bundles/preview/index.rst:
```diff
@@ -0,0 +1,117 @@
+Example
+-------
+
+This example is a simple (and not complete) example of a news-entity form.
```
`This is a simple (and not complete) example...`

danrot commented on bundles/preview/index.rst:
```diff
@@ -0,0 +1,117 @@
+                this.render();
+
+                this.sandbox.on('sulu.toolbar.save', this.save.bind(this));
+                this.options.preview.bindDomEvents(this.$el);
```
Is the preview really passed as an option?

wachterjohannes replied:
no only in the example i have copied from. i will fix it

danrot commented on bundles/route/index.rst:
```diff
@@ -0,0 +1,132 @@
+===========
+
+The RouteBundle provides the feature of managing website-routes for
+custom-entities. This routes will be generated by calling the
```
`These routes`

danrot commented on bundles/route/index.rst:
```diff
@@ -0,0 +1,132 @@
+    }
+
+Register this class as a service with the tag `<tag name="sulu_route.defaults_provider"/>`
+and call on creating a new entity the `save` method of the service
```
Don't really know what you want to say with that sentence.

danrot commented:
Looks like this PR also includes https://github.com/sulu/sulu-docs/pull/245/files

wachterjohannes commented:
@danrot yes i say that we merge this and close the other one.

wachterjohannes commented:
@danrot i have fixed all the issue and make the last sentence of the route-bundle clearer. additionally i have added a not which mentions the `sulu:route:update` command

danrot commented on bundles/route/index.rst:
```diff
@@ -0,0 +1,139 @@
+
+    To update already existing entities you can run the command
+    `console sulu:route:update AppBundle\Entity\Recipe` which updates or create the route
+    (if no exists) for all the entities of this type.
```
remove `(if not exists)`, should be clear with `updates or creates`

danrot commented on bundles/route/index.rst:
```diff
@@ -0,0 +1,139 @@
+.. note::
+
+    To update already existing entities you can run the command
+    `console sulu:route:update AppBundle\Entity\Recipe` which updates or create the route
```
`app/console` (I know about sulu-minimal, but only `console` won't work anywhere)
`updates or creates`

wachterjohannes commented:
@danrot fixed
